src/boot/efi/pe.c

```diff
@@ -153,15 +155,32 @@ static void locate_sections(
         assert(offsets);
         assert(sizes);
 
+        EFI_STATUS err;
+
         for (size_t i = 0; i < n_table; i++) {
                 const PeSectionHeader *sect = section_table + i;
 
                 for (size_t j = 0; sections[j]; j++) {
                         if (memcmp(sect->Name, sections[j], strlen8(sections[j])) != 0)
                                 continue;
 
-                        offsets[j] = in_memory ? sect->VirtualAddress : sect->PointerToRawData;
-                        sizes[j] = sect->VirtualSize;
+                        size_t offset = in_memory ? sect->VirtualAddress : sect->PointerToRawData;
+                        size_t size = sect->VirtualSize;
+
+                        offsets[j] = offsets[j] ? : offset;
+                        sizes[j] = sizes[j] ? : size;
```

Comment:
uh, this is weird. if for some reason a section has a zero size you'll end up filling in the size of a different section here.